slack-plugin/src/main/kotlin/slack/gradle/SlackTools.kt

```diff
@@ -98,91 +91,19 @@ public abstract class SlackTools @Inject constructor(providers: ProviderFactory)
     }
   }
 
-  private val thermalsLogProcess: Process? =
-    if (logThermals && AppleSiliconCompat.Arch.get() == X86_64) {
-      ProcessBuilder("pmset", "-g", "thermlog").redirectOutput(thermalsFile()).start()
-    } else {
-      null
-    }
-
-  // TODO unify this API with the other thermals logging?
-  private val m1Thermals: BehaviorSubject<Thermals> = BehaviorSubject.create()
-  @Suppress("UNCHECKED_CAST")
-  private val m1ThermalsHeartbeat: Disposable =
-    if (logThermals && AppleSiliconCompat.Arch.get() == ARM64) {
-      Observable.interval(5, SECONDS)
-        .map {
-          ThermLog(
-            timestamp = LocalDateTime.now(),
-            schedulerLimit = 0,
-            availableCpus = 0,
-            // These aren't entirely true numbers but best effort for now
-            speedLimit =
-              when (M1ThermalParser.getThermals()) {
-                NOMINAL -> 100
-                FAIR -> 75
-                SERIOUS -> 50
-                CRITICAL -> 25
-                null -> -1
-              },
-          )
-        }
-        .filter { it.speedLimit != -1 }
-        .map {
-          thermalsFile().appendText("\n${it.timestamp} - ${it.speedLimit}")
-          it
-        }
-        .scanWith<Thermals>({ Thermals.Empty }) { acc, next ->
-          when (acc) {
-            is Thermals.Empty -> ThermalsData(listOf(next))
-            is ThermalsData -> ThermalsData(acc.logs + next)
-          }
-        }
-        .subscribe { m1Thermals.onNext(it) }
-    } else {
-      Disposable.empty()
-    }
-
-  private var thermalsAtClose: Thermals? = null
-
-  /** Returns the current or latest captured thermals log. */
-  public val thermals: Thermals?
-    get() {
-      return thermalsAtClose ?: readThermalsResult()
-    }
-
-  @Synchronized
-  private fun readThermalsResult(): Thermals? {
-    return if (logThermals) {
-      if (AppleSiliconCompat.Arch.get() == ARM64) {
-        m1Thermals.value
-      } else {
-        val file = parameters.thermalsOutputFile.asFile.get()
-        // File may no longer exist if they were running `clean`
-        if (file.exists()) {
-          val thermalsLog = file.readText()
-          ThermalsParser.parse(thermalsLog)
-        } else {
-          Thermals.Empty
-        }
-      }
-    } else {
-      null
-    }
+  private fun peekThermals(): Thermals? {
+    return thermalsWatcher?.peek()
   }
 
   override fun close() {
-    m1ThermalsHeartbeat.dispose()
-    try {
-      // Close thermals process and save off its current value
-      thermalsAtClose = readThermalsResult()
-      if (!parameters.offline.get()) {
-        thermalsAtClose?.let { thermalsReporter?.reportThermals(it) }
-      }
-    } catch (t: Throwable) {
+    // Close thermals process and save off its current value
+    thermalsAtClose = thermalsWatcher?.stop()
+    if (!parameters.offline.get()) {
+      thermalsAtClose?.let { thermalsReporter?.reportThermals(it) }
+    }
+    try {} catch (t: Throwable) {
```

Comment:
Try block here is empty. Did you mean to put the lines of code above to this try block? Otherwise, we should just remove it.

Reply:
Oops, yeah that's a leftover

Reply:
[652e92a](https://github.com/slackhq/slack-gradle-plugin/pull/7/commits/652e92a801c377751ffb61bcd493bb77f38d2e55)